### PR TITLE
feat(core:api): expose endpoint for pagination for objects in a job

### LINKED ISSUE #61: 2. Route for getting objects with pagination - core

### PR DESCRIPTION
**Pull request by tomrtk**
- feat(core:api): expose endpoint for pagination for objects in a job. Closing #61.
- test(core): refactor db fixture to use main setup and shutdown functions

### CODE REVIEW

File: src/core/api/api.py [hunk 1]
```diff
@@ -611,3 +611,43 @@ async def get_object_image(
...
+    start: int = Query(0, ge=0),
+    lenght: int = Query(10, ge=1),
+) -> Dict[str, Any]:
+    """Endpoint to get part of objects from a job.
+
+    Returns
+    -------
+    Dict[str, Any]
+        Containing total number of object in job and part of objects from
+        `start` to `start + lenght`.
```
Comment (MindTooth) [inline src/core/api/api.py]:
```suggestion
        `start` to `start + length`.
```

File: src/core/api/api.py [hunk 2]
```diff
@@ -611,3 +611,43 @@ async def get_object_image(
...
+        raise HTTPException(503, repr(e))
+
+    if data is not None:
+        # convert to schema Object's:
+        data["data"] = [schema.Object(**o.to_api()) for o in data["data"]]
+        return data
+    else:
+        msg = f"project with id {project_id} or job with id {job_id} not found"
+        logger.warning(msg)
+        raise HTTPException(422, msg)
```
Comment (MindTooth) [inline src/core/api/api.py]:
```suggestion
    if data is None:
        msg = f"project with id {project_id} or job with id {job_id} not found"
        logger.warning(msg)
        raise HTTPException(422, msg)
        
     # convert to schema Object's:
     data["data"] = [schema.Object(**o.to_api()) for o in data["data"]]
     return data
```

? 😉

File: src/core/services.py [hunk 3]
```diff
@@ -508,3 +508,56 @@ def queue_job(project_id: int, job_id: int, session: Session) -> None:
...
+        return None
+
+    job = project.get_job(job_id)
+
+    if job is None:
+        return None
+
+    response: Dict[str, Any] = {}
+    response["total_objects"] = len(job._objects)
+    response["data"] = job._objects[start : start + lenght]
```
Comment (MindTooth) [inline src/core/services.py]:
```suggestion
    response["data"] = job._objects[start : start + length]
```

File: src/core/services.py [hunk 4]
```diff
@@ -508,3 +508,56 @@ def queue_job(project_id: int, job_id: int, session: Session) -> None:
         logger.error(
             f"Cannot queue job {job_id}, it's of status {job.status()}."
         )
+
+
+def get_job_objects(
+    project_id: int, job_id: int, start: int, lenght: int
```
Comment (MindTooth) [inline src/core/services.py]:
```suggestion
    project_id: int, job_id: int, start: int, length: int
```

File: tests/unit/core/unit/test_service.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    result = get_job_objects(2, 1, 0, 1)
+    assert result is None
+
+    result = get_job_objects(1, 2, 0, 1)
+    assert result is None
+
+
+def test_get_job_objects_wrong_start_lenght(make_test_project_repo):
+    """Test `get_job_objects` with to large `start` and `lenght`."""
```
Comment (MindTooth) [inline tests/unit/core/unit/test_service.py]:
```suggestion
    """Test `get_job_objects` with to large `start` and `length`."""
```

**Review by R0flcopt3r — APPROVED** [review]

**Pull request merged**